### PR TITLE
Move cmake_test/ into cmake/

### LINKED ISSUE #74: Move 'cmake_test/' into 'cmake/'
**Is your feature request related to a problem? Please describe.**
CMakeTest has a layout that differs from the rest of CMakePP org repositories. They have CMake source files contained under the `cmake/` directory.

**Describe the solution you'd like**
The `cmake_test` directory should be moved inside the `cmake` directory.

### PR DESCRIPTION
**Pull request by AutonomicPerfectionist**
**Is this pull request associated with an issue(s)?**
Fixes #74 

**Description**
This PR moves the directory structure around a bit to more closely align with the standard used by other CMakePP projects. That means the top-level `cmake_test/` directory was moved underneath the `cmake/` directory, with module paths updated in `CMakeLists.txt`. The contents of the `tests/` directory was also moved under a new `tests/cmake_test` subdirectory, except for the tests' `CMakeLists.txt` file. The `test_directory_2` directory was then moved underneath `tests/`. `tests/CMakeLists.txt` has been updated with the new paths.

**TODOs**

- [x] More extensive testing with CMakePPLang and CMaize to ensure no regressions

### CODE REVIEW

**Comment by AutonomicPerfectionist** [comment]
@zachcran I've tested this branch with CMakePPLang and CMaize, and both still pass all of their tests. Are there any other projects we should run against it before merging?

**Comment by zachcran** [comment]
CMakePPLang and CMaize should be sufficient!

File: docs/source/conf.py [hunk 1]
```diff
@@ -109,7 +109,7 @@
 
 # -- Generate documentation -------------------------------------------------
 
-examples_dir = os.path.join(root_path, "tests", "tutorials")
+examples_dir = os.path.join(root_path, "tests/cmake_test", "tutorials")
```
Comment (zachcran) [inline docs/source/conf.py:112]: We should let `os.path.join` determine the path separators here:
```suggestion
examples_dir = os.path.join(root_path, "tests", "cmake_test", "tutorials")
```

File: docs/source/conf.py [hunk 2]
```diff
@@ -50,7 +50,7 @@
 ########################################
 
 cminx_out_dir = os.path.join(src_dir, "developer", "cmake_test")
-cminx_in_dir = os.path.join(root_path, "cmake_test")
+cminx_in_dir = os.path.join(root_path, "cmake/cmake_test")
```
Comment (zachcran) [inline docs/source/conf.py:53]: We should let os.path.join determine the path separators here:
```suggestion
cminx_in_dir = os.path.join(root_path, "cmake", "cmake_test")
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -19,6 +19,6 @@ if("${BUILD_TESTING}")
     message("Building tests")
 
     include(CTest)
-    add_subdirectory("tests")
+    add_subdirectory("tests/")
```
Comment (zachcran) [inline CMakeLists.txt:22]: For my own edification, is the `/` here necessary, or is it just to make things more explicit?

**Review by zachcran — CHANGES_REQUESTED** [review]
This looks good. I just had a few small revisions and a question.

**Review by ryanmrichard — APPROVED** [review]

File: CMakeLists.txt (hunk 3 above)
Comment (ryanmrichard) [reply CMakeLists.txt:22]: TBH I also don't know about this particular case, but I'm pretty sure it matters in other scenarios. IIRC, asking CMake to install `"x"` into a directory `y` will result in `y/x`, but asking it to install `"x/"` into a directory `y` will result in `y/contents_of_x`, *i.e.* the slash determines whether or not the directory and the contents are copied to `y` or if just the contents are copied.

**Review by zachcran — APPROVED** [review]

**Comment by zachcran** [comment]
FYI this is an API breaking change, so technically the major version should be incremented. Since we are still in an alpha/beta release, though, it is probably a bump to the minor version.

**Pull request merged**